### PR TITLE
colocated workspaces: tests and documentation

### PR DESCRIPTION
**Pull request by sjawhar**
## Summary

Adds comprehensive tests and documentation for the colocated workspaces feature stack.

**Key changes:**
- Add design document `docs/design/colocated-workspaces.md` covering:
  - Detection algorithm and corner cases
  - Warning scenarios and messages
  - Command behavior specifications
  - Known architectural limitations
- Add integration tests for:
  - `workspace add --colocate` (basic, error cases, git worktree creation)
  - Import from secondary worktree HEADs
  - `workspace forget` with git worktree cleanup
  - Symlink handling edge cases

## Test coverage

New tests:
- `test_colocated_workspace_git_symlink_to_wrong_repo`
- `test_workspace_add_colocate_basic`
- `test_workspace_add_colocate_non_colocated_error`
- `test_workspace_add_colocate_creates_git_worktree`
- `test_workspace_add_colocate_git_failure`
- `test_import_detects_secondary_worktree_head_change`
- `test_import_all_worktrees_heads`
- `test_workspace_forget_removes_git_worktree`
- `test_workspace_forget_with_custom_name_removes_git_worktree`
- `test_workspace_forget_handles_missing_worktree`
- `test_workspace_forget_non_colocated_no_git_cleanup`

## PR Stack

This PR is part of a stack implementing colocated workspaces (#8052):
1. #8642 - Base colocation support
2. #8644 - `workspace add --colocate`
3. #8645 - Import commits from all worktrees' HEADs
4. #8646 - Clean up Git worktree on `workspace forget`
5. **This PR** - Tests and documentation

### CODE REVIEW

File: docs/design/colocated-workspaces.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+### Per-workspace git_head Tracking
+
+Store `git_head` per workspace in the view, allowing accurate HEAD tracking
+across multiple colocated workspaces.
+
+### `jj workspace add --colocate`
+
+Automatically create a Git worktree when adding a jj workspace in a colocated
+repo, maintaining the colocated state.
```
Comment (Enzime) [inline docs/design/colocated-workspaces.md]: This would be nice to have in https://github.com/jj-vcs/jj/pull/8644

File: docs/design/colocated-workspaces.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+| Orphan worktrees (unborn branch) | Unknown | Worktree on unborn branch | Low |
+
+### Command Behavior Corner Cases
+
+#### `jj workspace add`
+
+| Scenario | Current Behavior | Expected Behavior |
+|----------|------------------|-------------------|
+| In colocated repo | Creates jj workspace only | Future: `--colocate` option to create Git worktree too |
+| In non-colocated repo | Creates jj workspace | Correct |
```
Comment (Enzime) [inline docs/design/colocated-workspaces.md]: What does Correct mean for the expected behaviour?

File: docs/design/colocated-workspaces.md
Comment (PhilipMetzger) [inline docs/design/colocated-workspaces.md:1]: nit: A design doc is usually a separate discussion and since the feature already existed without it, I don't think its necessary at all.

File: docs/design/colocated-workspaces.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
+# Colocated Workspaces
+
+Author: [sjawhar](https://github.com/sjawhar) (sami@metr.org)
```
Comment (PhilipMetzger) [inline docs/design/colocated-workspaces.md:3]: nit: use the usual formatting for this reason there's an actual blueprint

**Review by PhilipMetzger — CHANGES_REQUESTED** [review]
you should add the tests to the commits which implement feature and ditto with documentation. 

Also design docs aren't documentation if anything, they're technical discussions about potentially large features which may have an impact on core jj.
